src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php

```diff
@@ -118,6 +118,10 @@ public function reverseTransform($value)
             throw new TransformationFailedException('Expected a string.');
         }
 
+        if (true === str_contains($value, "\0")) {
+            throw new TransformationFailedException('Null bytes not allowed');
```

Comment:
```suggestion
            throw new TransformationFailedException(sprintf('The value contains null bytes: "%s".', $value));
```

Reply:
@smnandre 

I tried to follow a bit the pattern of the error thrown by the DateTime class, which does not display the value of the injected field. Would it be correct to display it without escaping?

Reply:
I don't think we should deliberately inject null bytes into an exception message.